library/model/book.py

```diff
@@ -6,146 +6,172 @@
-            return 5
-        elif self._book_type == "Electronic":
-            return 2
-        elif self._book_type == "Audio":
-            return 2
-        else:
-            raise AttributeError("No such book type...")
+        pass
 
+    @abstractmethod
     def borrow_book(self) -> "BorrowedBook":
-        if self.can_borrow():
-            if self._book_type == "Paper":
-                self.borrowed_items += 1
-            LibraryRepository.update_book(self)
-            borrowed_book = BorrowedBook.from_book(self)
-            borrowed_book.due_date = datetime.now() + timedelta(days=7)
-            borrowed_book.current_fee = self.get_weekly_fee()
-            return borrowed_book
-        raise ValueError("Book cannot be borrowed")
+        pass
+
+    def _create_book_borrow(self) -> "BorrowedBook":
+        LibraryRepository.update_book(self)
+        borrowed_book = BorrowedBook(self)
+        return borrowed_book
+
+    @abstractmethod
+    def return_book(self):
+        pass
+
+    def _handle_book_return(self):
+        LibraryRepository.update_book(self)
 
     def __eq__(self, other):
         """Overrides the default implementation"""
         if isinstance(other, Book) or isinstance(other, BorrowedBook):
-            return self.isbn == other.isbn and self._book_type == other._book_type
+            return self.isbn == other.isbn and (type(self) is type(other))
         return NotImplemented
 
     def __str__(self):
-        return BookSerializer().serialize(self, "JSON")
+        return BookSerializer.serialize(self, "JSON")
+
+class AudioBook(Book):
+    duration: int = 0
+
+    def __init__(self, title, authors, publisher, pub_date, genres, isbn, duration):
+        Book.__init__(self, title, authors, publisher, pub_date, genres, isbn)
+        self.duration = duration
+
+    def can_borrow(self) -> bool:
+        return True
+
+    def borrow_book(self) -> "BorrowedBook":
+        if self.can_borrow():
+            return self._create_book_borrow()
+        raise ValueError("Book cannot be borrowed")
+
+    def return_book(self):
+        self._handle_book_return(self)
+
+    def get_approximate_duration(self) -> int:
+        return self.duration
+
+    def get_weekly_fee(self) -> int:
+        return 2
+
+class ElectronicBook(Book):
+    pages: int
+
+    def __init__(self, title, authors, publisher, pub_date, genres, isbn, pages):
+        Book.__init__(self, title, authors, publisher, pub_date, genres, isbn)
+        self.pages = pages
+    def can_borrow(self) -> bool:
+        return True
+
+    def borrow_book(self) -> "BorrowedBook":
+        if self.can_borrow():
+            return self._create_book_borrow()
+        raise ValueError("Book cannot be borrowed")
 
+    def return_book(self):
+        self._handle_book_return(self)
+
+    def get_approximate_duration(self) -> int:
+        return self.pages * 5 * 60
 
-class BorrowedBook(Book):
+    def get_weekly_fee(self) -> int:
+        return 2
```

Comment:
Please use a variable for the fee, because like this the fee is not changeable in future changes.